app/src/main/java/net/gsantner/opoc/frontend/filebrowser/GsFileBrowserListAdapter.java

```diff
@@ -840,25 +831,28 @@ private StringFilter(GsFileBrowserListAdapter adapter, List<File> adapterData) {
         @Override
         protected FilterResults performFiltering(CharSequence constraint) {
             final FilterResults results = new FilterResults();
-            constraint = constraint.toString().toLowerCase(Locale.getDefault()).trim();
-            _filteredList.clear();
 
-            if (constraint.length() == 0) {
-                _filteredList.addAll(_originalList);
-            } else {
-                for (File file : _originalList) {
-                    if (file.getName().toLowerCase(Locale.getDefault()).contains(constraint)) {
-                        _filteredList.add(file);
-                    }
-                }
-            }
+            _lastFilter = constraint.toString().toLowerCase().trim();
+            _filter(_originalList, _filteredList);
```

Comment:
we don't want to run an async filter inside load folder as it is already async. So we break the internal processing out into a stand-alone function